lib/TimezoneGuesser/GussFromCustomizedTimeZone.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sabre\VObject\TimezoneGuesser;
+
+use DateTimeZone;
+use DateTimeImmutable;
+use Sabre\VObject\Component\VTimeZone;
+use Sabre\VObject\Recur\RRuleIterator;
+use Sabre\VObject\TimeZoneUtil;
+
+class GussFromCustomizedTimeZone implements TimezoneGuesser
+{
+    public function guess(VTimeZone $vtimezone, bool $failIfUncertain = false): ?DateTimeZone
+    {
+        if ($vtimezone->TZID->getValue() !== 'Customized Time Zone') {
+            return null;
+        }
+
+        $timezones = DateTimeZone::listIdentifiers();
+        $standard = $vtimezone->STANDARD;
+        $daylight = $vtimezone->DAYLIGHT;
+
+        $standardOffset = $standard->TZOFFSETTO->getValue();
+        $standardRRule = $daylight ? $standard->RRULE->getValue() : 'FREQ=DAILY';
+        $year = (new DateTimeImmutable('now'))->format('Y');
+        $start = new DateTimeImmutable($year . '-01-01');
```

Comment:
With this solution, we will check that the timezone match in the current year transitions but not necessarily when the event using this timezone is happening. For eg, if the event is happening in 1990, we could find a timezone match in 2021 but the timezone definition for this match might be different when we use it to extract the offsets back in 1990.

It will likely be difficult to have a perfect solution here, however, could you document this known issue in a comment? Do you think that having a parameter to turn on or off this guesser would make sense? Since the match is not guaranteed to be 100% accurate, we could turn this guesser off and fail if needed

Reply:
Added the comments. I think we can wait for this situation happens and then to see how do we want to handle it